### ISSUE TITLE
some of questions regarding the tool

### ISSUE THREAD

**Issue by ofiryaish** [johli/scrambler#2]
Hello, 
Thank you for your work. I have some questions about the tool:

1) Is there an option to use this tool with a model trained on TensorFlow 2? 
2) does the tool give negative attributions scores as other tools as Integrated gradient does?

Thank you,
Ofir

**Comment by jlinder2** [comment]
Dear Ofir,

Sorry for the late response (currently on vacation). Thank you for your interest in using the tool!

1. Unfortunately the tool only supports Tensorflow 1 models (see the reqs in the readme file). For many models with standard layers, it's easy to convert them from Tensorflow 2 to 1. There are a few guides online; one of the simplest ways is to simply re-initialize the Model object with the full definition of layers in Keras (with Tensorflow 1) and then call the load_weights function on the Model object (https://keras.io/api/models/model_saving_apis/#loadweights-method) with the .h5 weights file from Tensorflow 2 as input argument.

2. Yes, the tool does support attributions similar in concept to negative and positive scores. Specifically, rather than finding maximally reconstructive features, we can optimize the scrambler to find maximally repressive (negative features). To use this functionality, you can start off from the APA example code (https://github.com/johli/scrambler/blob/master/examples/dna/scrambler_apa_example.ipynb), but instead of calling the scrambler.train function with arguments nll_mode='reconstruction' and predictor_task='classification_sym', you should use nll_mode='minimization' (this is to find negative scores; use 'maximization' to find positive scores) and either predictor_task='classification' or predictor_task='regression' (depending on whether your predictor model is a classifier or a regressor).

Hopefully these answers were helpful to you. If not (or if something's not working properly) please get back to me and I'll try and give more help.

Best,

Johannes
